cognite_toolkit/_cdf_tk/templates/_templates.py

```diff
@@ -103,186 +104,280 @@ def build_config(
-            sql_file2 = filepath_src.parent / f"{ext_id}.sql"
-            if not sql_file2.exists():
-                print("      [bold yellow]WARNING:[/] could not find sql file:")
-                print(f"                 [bold]{sql_file1.name}[/] or ")
-                print(f"                 [bold]{sql_file2.name}[/]")
-                print(f"               Expected to find it next to the yaml file at {sql_file1.parent}.")
-                return False
-        parts = ext_id.split("_")
-        if len(parts) < 2:
-            print(
-                f"      [bold yellow]WARNING:[/] the transformation {filepath_src} has an externalId [bold]{ext_id}[/] without the recommended '_' based namespacing."
-            )
-        elif parts[0] != "tr":
-            print(
-                f"      [bold yellow]WARNING:[/] the transformation {filepath_src} has an externalId [bold]{ext_id}[/] without the recommended 'tr_' based prefix."
-            )
-    elif resource_type == "data_models" and ext_id_type == "space":
-        parts = ext_id.split("_")
-        if len(parts) < 2:
+        else:
             print(
-                f"      [bold yellow]WARNING:[/] the space {filepath_src} has an externalId [bold]{ext_id}[/] without the recommended '_' based namespacing."
+                f"      {WARN_YELLOW} the group {filepath_src} has a name [bold]{ext_id}[/] without the "
+                "recommended '_' based namespacing."
             )
-        elif parts[0] != "sp":
-            if ext_id == "cognite_app_data" or ext_id == "APM_SourceData" or ext_id == "APM_Config":
-                if verbose:
-                    print(
-                        f"      [bold green]INFO:[/] the space {ext_id} does not follow the recommended '_' based namespacing because Infield expects this specific name."
-                    )
-            else:
+    elif parts[0] != "gp":
+        print(
+            f"      {WARN_YELLOW} the group {filepath_src} has a name [bold]{ext_id}[/] without the "
+            "recommended `gp_` based prefix."
+        )
+
+
+def _check_yaml_semantics_transformations_schedules(ext_id: str, filepath_src: Path, verbose: bool) -> None:
+    # First try to find the sql file next to the yaml file with the same name
+    sql_file1 = filepath_src.parent / f"{filepath_src.stem}.sql"
+    if not sql_file1.exists():
+        # Next try to find the sql file next to the yaml file with the external_id as filename
+        sql_file2 = filepath_src.parent / f"{ext_id}.sql"
+        if not sql_file2.exists():
+            print(f"      {WARN_YELLOW} could not find sql file:")
+            print(f"                 [bold]{sql_file1.name}[/] or ")
+            print(f"                 [bold]{sql_file2.name}[/]")
+            print(f"               Expected to find it next to the yaml file at {sql_file1.parent}.")
+            raise ToolkitYAMLFormatError
```

Comment:
We now supports sql inside the transformation file. But when adding that support, must have missed this